practice-app/server/src/tests/users/users.test.js

```diff
@@ -0,0 +1,292 @@
+                        email: wrong_emails[wEmail],
+                        password: "Password*11",
+                    })
+                    .expect(400);
+                }
+            });
+        });
+        describe("given incorrect password was provided", () => {
+            it("should return a 400", async () => {
+                wrong_passwords = ["password","Password","password*","password*12","Password11","Password*"]
+                for (const i in wrong_passwords) {
+                    await supertest(app)
+                    .post("/users/register")
+                    .send({
+                        first_name: "ersoy",
+                        last_name: "ersoy",
+                        email: "kadir@gmail.com",
+                        password: wrong_passwords[i],
+                    })
+                    .expect(400);
+                }
+            });
+        });
+        describe("given correct body was provided", () => {
+            it("should return a 201 with correct response", async () => {
+                axios.post.mockResolvedValueOnce({
+                    data: {
+                        access_token: "access_token",
+                    },
+                });
+                axios.post.mockResolvedValueOnce({
+                    data: {
+                        created_at: "created_at",
+                    },
+                });
+
+                const { body, statusCode } = await supertest(app)
+                    .post("/users/register")
+                    .send({
+                        first_name: "kadir",
+                        last_name: "ersoy",
+                        email: "kadir@gmail.com",
+                        password: "Password*11",
+                    });
+
+                expect(statusCode).toBe(201);
+                expect(body).toEqual({
+                    created_at: "created_at",
+                    message: `Created the user with kadir@gmail.com`,
+                });
+                const user_test = new User({ client_id:"id",email: 'kadir@gmail.com', given_name:"kadir",family_name:"ersoy",password:"Password*11" });
+                user_test.save();
+            });
+        });
+        describe("given already existing user was provided", () => {
+            it("should return a 409", async () => {
+                axios.post.mockResolvedValueOnce({
+                    data: {
+                        access_token: "access_token",
+                    },
+                });
+                axios.post.mockRejectedValueOnce({
+                    response: {
+                        status: 409,
+                        data:{
+                            statusCode: 409,
+                            error: "Conflict",
+                            message: 'The user already exists.',
+                            errorCode: 'auth0_idp_error' 
+                        }
+                    }
+                });
```

Comment:
Why are there two mocks for axios.post

Reply:
There is two axios request for register endpoint. One for admin token and one for creating user.